Pin the versions of the packages that are used in tests and add dependabot (#37)
CODAIT/pydax: CONTRIBUTING.md

1. Install `tox`:

    $ pip install -U -r requirements/tox.txt  # If you are inside a virtual environment, conda environment

>     $ pip3 install --user -U tox  # If you are outside any virtual environment or conda environment

Reviewer: Do we want to install from `requirements/tox.txt` when outside of a virtual env?

Author: Outside an virtual environment is sort of a user's global installation that we better not impose such a restriction though

Author: I updated this line to install our specified version, but appended additional note to the comment (no need to run this line if it's a user's global installation)

Reviewer: gotcha sounds good

The reST code style compliance is also checked by the `tox` lint test environment if you prefer to use that:

    tox -e lint

## Dependency Version Pinning Policy

We should pin the versions of all Python packages that we are using solely for testing and doc
generating for a stable test and doc env (e.g., future incompatibility, regression, etc.). We want

> to pin these because, in this project, We use these packages soly for deployment of our development

Reviewer: ```suggestion
to pin these because, in this project, we use these packages solely for deployment of our development
```

environment (i.e., running tests and generating docs) and we want stable packages that are used by

> us for these purposes. We should let some dependency bot to verify that bumping the versions won't

Reviewer: ```suggestion
us for these purposes. We let GitHub's Dependabot verify that bumping the versions won't
```
